lib/xclarity_client/scalable_complex.rb

```diff
@@ -3,12 +3,11 @@ class ScalableComplex
     include XClarityClient::Resource
 
     BASE_URI = '/scalableComplex'.freeze
-    LIST_NAME = 'complex'.freeze
+    LIST_NAME = ['complex', 'complexList'].freeze
```

Comment:
`.freeze` doesn't works in Ruby Lists. It didn't the list contents freezed, for example:

```
irb(main):001:0> LIST_NAME = ['complex', 'complexList'].freeze
=> ["complex", "complexList"]
irb(main):002:0> LIST_NAME[0][1] = 'A'
=> "A"
irb(main):003:0> LIST_NAME
=> ["cAmplex", "complexList"]
```
If you really want immutable elements in your `LIST_NAME`, you can do something like:

```
irb(main):001:0> LIST_NAME = ['complex', 'complexList'].map{ |element| element.freeze }.freeze
=> ["complex", "complexList"]
irb(main):002:0> LIST_NAME[0][1] = 'A'
RuntimeError: can't modify frozen String
	from (irb):2:in `[]='
	from (irb):2
	from /Users/walter/.rvm/rubies/ruby-2.4.1/bin/irb:11:in `<main>'
```

Reply:
The point here is that all standards in this project (and in other ManageIQ providers also) are using strings, not lists. Then you can not use it as a basis for what you are doing. Once the elements inside the list can be changed, I recommend you to ensure that it will not happen.

Reply:
Well, not really. There's some frozen lists in MiQ project, but, yeah, you're right, it could be done by the right way like you said